src/tools/UpdatePackageDependencies.ts

```diff
@@ -155,7 +137,55 @@ export async function updatePackageDependencies() : Promise<void> {
     fs.writeFileSync('package.json', content);
 }
 
-function getLowercaseFileNameFromUrl(url : string) : string {
+
+function replaceVersion(value: string, newVersion: string): string {
+    if (!value) {
+        return value; //if the value is null or undefined return the same one
+    }
+
+    let regex: RegExp = dottedVersionRegExp;
+    let newValue: string = newVersion;
+    if (!dottedVersionRegExp.test(value)) {
+        regex = dashedVersionRegExp;
+        newValue = newVersion.replace(/\./g, "-");
+    }
+    dottedVersionRegExp.lastIndex = 0;
+
+    if (!regex.test(value)) {
+        return value; //If the string doesn't contain any version return the same string
+    }
+
+    return value.replace(regex, newValue);
+}
+
+function verifyMatchCount(value: string, shouldContainVersion = false): void {
```

Comment:
nit: If I didn't write the code, I am not sure I would understand what "verifyMatchCount" does without reading the code. How about "verifyVersionSubstringCount"?